shell/platform/windows/flutter_windows_engine.cc

```diff
@@ -573,15 +580,22 @@ void FlutterWindowsEngine::RemoveView(FlutterViewId view_id) {
     info.view_id = view_id;
     info.user_data = &captures;
     info.remove_view_callback = [](const FlutterRemoveViewResult* result) {
-      // This is invoked on the raster thread, the same thread that the present
-      // callback is invoked. If |FlutterRemoveViewResult.removed| is `true`,
-      // the engine guarantees the view won't be presented.
+      // This is invoked on an engine thread. If
+      // |FlutterRemoveViewResult.removed| is `true`, the engine guarantees the
+      // view won't be presented.
       Captures* captures = reinterpret_cast<Captures*>(result->user_data);
       captures->removed = result->removed;
       captures->latch.Signal();
     };
 
-    embedder_api_.RemoveView(engine_, &info);
+    FlutterEngineResult result = embedder_api_.RemoveView(engine_, &info);
+    if (result != kSuccess) {
+      // Starting the remove view operation failed. This is unexpected and
+      // indicates a bug in the Windows embedder.
```

Comment:
If failures indicate bugs of the Windows embedder, should we just assert it?

Reply:
On the fence on this one. I was going to make the same comment exactly -- at least in debug mode, but I can see the argument for either approach depending on how severe the repercussions of the failure are.

Reply:
It feels unease to me that if somehow this branch is triggered due to a bug, the bug can exist on for months unreported because it's just a harmless log in the user's app, and the user might think "oh, right, I shouldn't add an implicit view".

Reply:
I switched to assertions for the result value. The engine does basic spot checks here (is it an implicit view ID? Did you provide invalid arguments?) that the Windows embedder should _never_ fail.